pkg/kclient/deployments.go

```diff
@@ -137,13 +138,13 @@ func (c *Client) WaitForPodNotReady(name string) error {
 			if !ok {
 				return errors.New("error getting value from resultchan")
 			}
-			if pod, ok := val.Object.(*corev1.Pod); ok {
-				for _, cond := range pod.Status.Conditions {
-					if cond.Type == "Ready" {
-						if cond.Status == corev1.ConditionFalse {
-							return nil
-						}
-					}
+			if val.Type == apiMachineryWatch.Deleted {
+				return nil
+			}
+
+			if val.Type == apiMachineryWatch.Added {
```

Comment:
In which circumstances can this happen? We are watching for pods with `metadata.name` being `name`